core/DataTable/Renderer/Xml.php

```diff
@@ -370,7 +370,11 @@ protected function renderDataTable($array, $prefixLine = "")
                 foreach ($row as $name => $value) {
                     // handle the recursive dataTable case by XML outputting the recursive table
                     if (is_array($value)) {
-                        $value = "\n" . $this->renderDataTable($value, $prefixLine . "\t\t");
+                        if (is_array(reset($value))) {
```

Comment:
can you add a (system) test for this maybe?

Reply:
Technically since it's used in the `<categories>` array, it's already tested in this PR. If possible I can add a unit test as well.